pennylane/plugins/expt_tensornet_tf.py

```diff
@@ -162,241 +189,183 @@ def CRot3(a, b, c):
+    when the number of parameters to be optimized is large.
+
+    To use this device, you will need to install TensorFlow and TensorNetwork:
+
+    .. code-block:: bash
+
+        pip install tensornetwork>=0.2 tensorflow>=2.0
+
+    In addition, you will need to use the new-style QNode to enable
+    calculation of the quantum gradient using TensorFlow.
+
+    **Example:**
+
+    >>> from pennylane.beta.qnodes import qnode
+    >>> dev = qml.device("expt.tensornet.tf", wires=1)
+    >>> @qnode(dev, interface="autograd", diff_method="best")
+    >>> def circuit(x):
+    ...     qml.RX(x[1], wires=0)
+    ...     qml.Rot(x[0], x[1], x[2], wires=0)
+    ...     return qml.expval(qml.PauliZ(0))
+    >>> grad_fn = qml.grad(circuit, argnum=[0])
+    >>> print(grad_fn([0.2, 0.5, 0.1]))
+    ([array(-0.22526717), array(-1.00864546), array(6.9388939e-18)],)
+
+    .. note::
+
+        TensorFlow is used as the device backend, and is independent
+        of the chosen QNode interface. In the example above, we combine
+        ```expt.tensornet.tf`` with the ``autograd`` interface.
+        It can also be used with the ``torch`` and the ``tf`` interface.
+
     Args:
         wires (int): the number of modes to initialize the device in
+        shots (int): the number of shots used for returning samples
     """
+
     # pylint: disable=too-many-instance-attributes
     name = "PennyLane TensorNetwork (TensorFlow) simulator plugin"
     short_name = "expt.tensornet.tf"
     _capabilities = {"model": "qubit", "tensor_observables": True, "provides_jacobian": True}
 
-    _operation_map = {
-        "QubitStateVector": None,
-        "BasisState": None,
-        "QubitUnitary": unitary,
-        "PauliX": X,
-        "PauliY": Y,
-        "PauliZ": Z,
-        "Hadamard": H,
-        "S": S,
-        "T": T,
-        "CNOT": CNOT,
-        "SWAP": SWAP,
-        "CSWAP": CSWAP,
-        "CZ": CZ,
-        "PhaseShift": Rphi,
-        "RX": Rotx,
-        "RY": Roty,
-        "RZ": Rotz,
-        "Rot": Rot3,
-        "CRX": CRotx,
-        "CRY": CRoty,
-        "CRZ": CRotz,
-        "CRot": CRot3,
-    }
-
-    # observable mapping is inherited from expt.tensornet
-
-    def __init__(self, wires, shots=1000, analytic=True):
-        super().__init__(wires, shots)
-        self.backend = "tensorflow"
-
+    _operation_map = copy.copy(TensorNetwork._operation_map)
+    _operation_map.update(
+        {
+            "PhaseShift": Rphi,
+            "RX": Rotx,
+            "RY": Roty,
+            "RZ": Rotz,
+            "Rot": Rot3,
+            "CRX": CRotx,
+            "CRY": CRoty,
+            "CRZ": CRotz,
+            "CRot": CRot3,
+        }
+    )
+
+    backend = "tensorflow"
+    reshape = staticmethod(tf.reshape)
+    array = staticmethod(tf.constant)
+    asarray = staticmethod(tf.convert_to_tensor)
+    real = staticmethod(tf.math.real)
+    imag = staticmethod(tf.math.imag)
+    abs = staticmethod(tf.abs)
+
+    C_DTYPE = C_DTYPE
```

Comment:
:man_facepalming:

Reply:
Oh actually I worked it out, it doesn't complain because it can somehow work out the difference in scope (setting a class variable with a module variable)